hpe3parclient/client.py

```diff
@@ -508,46 +508,59 @@ def createVolume(self, name, cpgName, sizeMiB, optional=None):
 
         """
         info = {'name': name, 'cpg': cpgName, 'sizeMiB': sizeMiB}
-        # For primera array there is no compression and tdvv fields
-        # removing tdvv and replacing compression with reduce field
+        # For primera array there is no compression and tdvv keys
+        # removing tdvv, compression and
+        # replacing compression+tdvv with reduce key for DECO
         if not optional and self.primera_supported:
-            optional = {}
-            optional['tpvv'] = True
+            optional = {'tpvv': True}
         if optional:
             if self.primera_supported:
-                if 'tdvv' in optional:
-                    optional.pop('tdvv')
-
-                if optional.get('tpvv') is True \
+                for key in ['tpvv', 'compression', 'tdvv']:
+                    option = optional.get(key)
```

Comment:
is it allowed to send tpvv or other keys as true/false ? If yes then these values needs to be converted to Boolean.